pkg/webhook/clusterpropagationpolicy/mutating_test.go

```diff
@@ -0,0 +1,218 @@
+				SpreadConstraints: []policyv1alpha1.SpreadConstraint{
+					{SpreadByLabel: "", SpreadByField: "", MinGroups: 0},
+				},
+				ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+					ReplicaSchedulingType: policyv1alpha1.ReplicaSchedulingTypeDivided,
+				},
+			},
+			PropagateDeps: false,
+			ResourceSelectors: []policyv1alpha1.ResourceSelector{
+				{
+					Kind:       util.ServiceImportKind,
+					APIVersion: mcsv1alpha1.GroupVersion.String(),
+				},
+			},
+			Failover: &policyv1alpha1.FailoverBehavior{
+				Application: &policyv1alpha1.ApplicationFailoverBehavior{
+					PurgeMode:          policyv1alpha1.Graciously,
+					GracePeriodSeconds: nil,
+				},
+			},
+		},
+	}
+
+	// Define the expected policy after mutations.
+	wantPolicy := &policyv1alpha1.ClusterPropagationPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: policyName,
+			Labels: map[string]string{
+				policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel: "some-unique-uuid",
+			},
+			Finalizers: []string{util.ClusterPropagationPolicyControllerFinalizer},
+		},
+		Spec: policyv1alpha1.PropagationSpec{
+			Placement: policyv1alpha1.Placement{
+				SpreadConstraints: []policyv1alpha1.SpreadConstraint{
+					{
+						SpreadByField: policyv1alpha1.SpreadByFieldCluster,
+						MinGroups:     1,
+					},
+				},
+				ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+					ReplicaSchedulingType:     policyv1alpha1.ReplicaSchedulingTypeDivided,
+					ReplicaDivisionPreference: policyv1alpha1.ReplicaDivisionPreferenceWeighted,
+				},
+				ClusterTolerations: []corev1.Toleration{
+					*helper.NewNotReadyToleration(notReadyTolerationSeconds),
+					*helper.NewUnreachableToleration(unreachableTolerationSeconds),
+				},
+			},
+			PropagateDeps: true,
+			ResourceSelectors: []policyv1alpha1.ResourceSelector{
+				{
+					Kind:       util.ServiceImportKind,
+					APIVersion: mcsv1alpha1.GroupVersion.String(),
+				},
+			},
+			Failover: &policyv1alpha1.FailoverBehavior{
+				Application: &policyv1alpha1.ApplicationFailoverBehavior{
+					PurgeMode:          policyv1alpha1.Graciously,
+					GracePeriodSeconds: ptr.To[int32](failOverGracePeriodSeconds),
+				},
+			},
+		},
+	}
+
+	// Mock decoder that decodes the request into the policy object.
+	decoder := &fakeMutationDecoder{
+		obj: policy,
+	}
+
+	// Marshal the expected policy to simulate the final mutated object.
+	wantBytes, err := json.Marshal(wantPolicy)
+	if err != nil {
+		t.Fatalf("Failed to marshal expected policy: %v", err)
+	}
+	req.Object.Raw = wantBytes
+
+	// Instantiate the mutating handler.
+	mutatingHandler := NewMutatingHandler(
+		notReadyTolerationSeconds, unreachableTolerationSeconds, decoder,
+	)
+
+	// Call the Handle function.
+	got := mutatingHandler.Handle(context.Background(), req)
+
+	// Verify that the only patch applied is for the UUID label. If any other patches are present, it indicates that the policy was not handled as expected.
+	if len(got.Patches) > 0 {
```

Comment:
Is the length of got.Patches only equal to 1?

Reply:
The logic of the previously incorporated code is the same. I have read it now and may need to be adjusted.

Reply:
Yes makes sense. I was thinking the same when working on unit testing the Work (#5532) PR to integrate it in Milestone 6 for refinement but let's do it now :)

Reply:
All is done :)

Reply:
> The logic of the previously incorporated code is the same. I have read it now and may need to be adjusted.

Everything is completed on both the current unmerged and merged PRs, with follow-up PRs :)